scanner/config/config.go

```diff
@@ -1,9 +1,143 @@
 package config
 
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	// DefaultConfiguration provides the default values for the scanner configuration.
+	DefaultConfiguration = Config{
+		HTTPListenAddr: ":9443",
+		GRPCListenAddr: ":8443",
+		Indexer: IndexerConfig{
+			Enable:       true,
+			DBConnString: "postgresql:///postgres?host=/var/run/postgresql",
+		},
+		Matcher: MatcherConfig{
+			Enable:       true,
+			DBConnString: "postgresql:///postgres?host=/var/run/postgresql",
+		},
+		// Default is empty.
+		MTLS: MTLSConfig{
+			CertsDir: "",
+		},
+		LogLevel: LogLevel(zerolog.InfoLevel),
+	}
+)
+
 // Config represents the Scanner configuration parameters.
 type Config struct {
-	// Mode specifies the mode in which Scanner will run.
-	//
-	// The options are: Indexer, Matcher, Combo.
-	Mode string `json:"mode"`
+	Indexer        IndexerConfig `yaml:"indexer"`
+	Matcher        MatcherConfig `yaml:"matcher"`
+	HTTPListenAddr string        `yaml:"http_listen_addr"`
+	GRPCListenAddr string        `yaml:"grpc_listen_addr"`
+	MTLS           MTLSConfig    `yaml:"mtls"`
+	LogLevel       LogLevel      `yaml:"log_level"`
+}
+
+func (c *Config) validate() error {
+	err := c.MTLS.validate()
+	if err != nil {
+		return fmt.Errorf("mtls: %w", err)
+	}
+	return nil
+}
+
+// IndexerConfig provides Scanner Indexer configuration.
+type IndexerConfig struct {
+	// Database provides indexer's database configuration.
+	DBConnString string `yaml:"db_conn_string"`
+	// Enable if false disables the Indexer service.
+	Enable bool `yaml:"enable"`
+	// GetLayerTimeout timeout duration of GET requests for layers
+	GetLayerTimeout Duration `yaml:"get_layer_timeout"`
```

Comment:
Do we want to set any default for this? For just a GET, maybe 30 seconds or so is a decent timeout? I'm thinking if we offer this option, we should set a default, as I can only imagine customers setting this themselves to make it longer (or infinite)